.github/workflows/tests.yml

```diff
@@ -38,74 +35,41 @@ jobs:
           && !contains(github.event.pull_request.labels.*.name, 'ci:run-full-test-suite')
           || false
          }}
-      pytest_store_durations: true
-
+  
   upload-stable-deps:
     needs: tests
-    runs-on: ubuntu-latest
-    if: ${{ github.event_name == 'push' }}
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
-          ref: master
-
-      - name: Prepare local repo
-        run: |
-          git config user.name "GitHub Actions Bot"
-          git config user.email "<>"
-          if git ls-remote --exit-code origin "refs/heads/${{ env.DEPS_BRANCH }}"; then
-            git checkout "${{ env.DEPS_BRANCH }}"
-          else
-            git checkout -b "${{ env.DEPS_BRANCH }}"
-          fi
-
-      - name: Download artifacts
-        uses: actions/download-artifact@v4
-        with:
-          pattern: frozen-*
-          path: .github/stable/
-          merge-multiple: true
-
-      - name: Determine if changes have been made
-        id: changed
-        run: |
-          echo "has_changes=$(git status --porcelain | wc -l | awk '{print $1}')" >> $GITHUB_OUTPUT
-
-      - name: Stage changes
-        if: steps.changed.outputs.has_changes != '0'
-        run: |
-          git add .github/stable/
-          git commit -m "Update changed dependencies"
-          git push -f --set-upstream origin "${{ env.DEPS_BRANCH }}"
+    uses: ./.github/workflows/check_in_artifact.yml
+    if: github.event_name == 'push'
+    with:
+      artifact_name_pattern: 'frozen-*'
+      artifact_save_path: '.github/stable/'
+      pull_request_head_branch_name: bot/stable-deps-update
+      commit_message_description: Frozen Dependencies Update
+      pull_request_title: Update stable dependency files
+      pull_request_body: |
+        Automatic update of stable requirement files to snapshot valid python environments.
 
-      # Create PR to master
-      - name: Create pull request
-        if: steps.changed.outputs.has_changes != '0'
-        uses: repo-sync/pull-request@v2
-        with:
-          source_branch: "${{ env.DEPS_BRANCH }}"
-          destination_branch: "master"
-          github_token: "${{ secrets.GITHUB_TOKEN }}"
-          pr_title: "Update stable dependency files"
-          # if updating the pr_body, please also update the same text in the docs.yml file
-          pr_body: |
-            Automatic update of stable requirement files to snapshot valid python environments.
+        Because bots are not able to trigger CI on their own, please do so by pushing an empty commit to this branch using the following command:
 
-            Because bots are not able to trigger CI on their own, please do so by pushing an empty commit to this branch using the following command:
+        ```
+        git commit --allow-empty -m 'trigger ci'
+        ```
+  
+  upload-durations-files:

```

Comment:
This job won't work correctly right now. The reason is that we always split the core, jax, and tensorflow tests among multiple runners. Since the original file name of the artifact being uploaded, and eventually downloaded, for each of the 3 interfaces is the same (eg. `core_tests_durations.json` for all core test durations artifacts), all artifacts but one will be lost.

We can see this in action in the PR that was opened by the workflow [here](https://github.com/PennyLaneAI/pennylane/pull/6332). You can see that each of the durations files have 5k, 7k, 2k lines respectively, when the original durations files have 28k, 16k, 5k lines respectively.

I think that the solution is one of the following:
* We update the `interface-unit-tests` workflow so that when running on schedule, none of the jobs for which we need to upload durations are split over runners. This would make it so that only a single artifact is generated that contains durations for _all_ the tests. However, this is inefficient, as we'll run tests that don't need to be run unnecessarily.
* We add a new workflow entirely that's very similar to `tests.yml`, that skips _all_ the tests except for core, tf, jax, and does not split tests among multiple jobs/runners. This will lead to code duplication though.
* We update the existing workflow to download files without overwriting previously downloaded artifacts, and use `jq` or some other json manipulation tool to collect all artifacts for each interface into a unified file. I don't know if this will be possible with just the `download-artifact` action and may need to be used in tandem with a step that renames the updated durations file before uploading it as an artifact. This, in my opinion, is the best option.

Let me know what your thoughts are.

Reply:
Great catch mudit, one of the goals of this PR is to reduce (or remove) code duplication, as it becomes a pain point to maintain in the future, so I don't want to do that .. let me explore a few options!

Reply:
I quite like the third option ... changing the name of the uploaded artifact is trivial as we can attach `${{ matrix.group }}` to the artifact name and now you have all completely unique names.

I will look into an efficient way to merge them

Reply:
Durations file handling has been removed from this PR and will be addresed separately